source/slang/core.meta.slang

```diff
@@ -1238,14 +1238,14 @@ extension matrix<T, R, C> : IDifferentiable
 //@ public:
 
     /// Sampling state for filtered texture fetches.
-__magic_type(SamplerState, $(int(SamplerStateFlavor::SamplerState)))
+__magic_type(SamplerStateType, $(int(SamplerStateFlavor::SamplerState)))
 __intrinsic_type($(kIROp_SamplerStateType))
 struct SamplerState
 {
 }
 
     /// Sampling state for filtered texture fetches that include a comparison operation before filtering.
-__magic_type(SamplerState, $(int(SamplerStateFlavor::SamplerComparisonState)))
+__magic_type(SamplerStateType, $(int(SamplerStateFlavor::SamplerComparisonState)))
```

Comment:
Unrelated to this PR: Passing a "flavor" through on these seems like overkill. We might as well just have the comparison and non-comparison cases be distinct AST node classes (with a common base).